### DISCUSSION TITLE
How to get involved

### DISCUSSION THREAD

**Discussion by RollsChris** [jesseduffield/lazygit#3093] [General]
Hi,

I'm looking to start learning go and I think this would be a good project to use as reference as I do.

Do you have any tips on how to get started?
like what sort of go app is this? are there any specific areas I should look at, design patterns your using?

any kind of gentle intro into understanding how lazygit works under the hood 😇

Thanks

**Comment by mark2185** [comment]
Welcome!

First thing I'd recommend is to _use_ `lazygit`, that's the easiest way you'll be able to grasp what it _is_, and what _isn't_. 

Here's a (slightly dated, some details may differ) tutorial of sorts, [15 lazygit features in under 15 minutes](https://m.youtube.com/watch?v=CPLdltN7wgE).

There are also showcases in the form of gifs regarding common operations in `lazygit` [here](https://github.com/jesseduffield/lazygit#features).

Take a look at issues with the `good first issue` tag ([click](https://github.com/jesseduffield/lazygit/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)) which are a good step towards your first contribution.

Feel free to ask for any pointers or clarifications regarding anything, either here on github or on discord!

**Reply by RollsChris** [reply]
Hey thanks for the reply,

I have been using lazygit for about a month now and I love it. 

I also understand git quite well, I suppose its more the fact of getting started with go so that I can understand the lazygit code base 😊.

Thanks.

**Comment by RollsChris** [comment]
Id really love a little explanation on the folder structure :) just a little deep dive in to how this GO project is structured... any ideas please :)

**Reply by mark2185** [reply]
This [video](https://www.youtube.com/watch?v=kNavnhzZHtk) is a bit longer, but it does go through the process of implementing a feature and explaining the structure and reasoning behind the project.
